app/src/examples/RuntimeTests/tests/animations/withSpring/variousConfig.test.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import Animated, { useSharedValue, useAnimatedStyle, withSpring } from 'react-native-reanimated';
+import React from 'react';
+import {
+  describe,
+  test,
+  expect,
+  mockAnimationTimer,
+  recordAnimationUpdates,
+  render,
+  wait,
+} from '../../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+import { Snapshots } from './withSpring.snapshot';
+import { SpringConfig } from '../../../../../../../lib/typescript/reanimated2/animation/springUtils';
+
+const AnimatedComponent = ({
+  animateFrom,
+  animateTo,
+  config,
+}: {
+  animateFrom: number;
+  animateTo: number;
+  config: SpringConfig;
+}) => {
+  const widthSV = useSharedValue(animateFrom);
+
+  const style = useAnimatedStyle(() => {
+    return {
+      width: withSpring(widthSV.value, config),
+    };
+  });
+
+  useEffect(() => {
+    widthSV.value = animateTo;
+  }, [widthSV, animateTo]);
+
+  return (
+    <View style={styles.container}>
+      <Animated.View style={[styles.animatedBox, style]} />
+    </View>
+  );
+};
+
+async function getSnapshotUpdates(animateFrom: number, animateTo: number, config: SpringConfig, waitTime = 2000) {
+  await mockAnimationTimer();
+  const updatesContainer = await recordAnimationUpdates();
+  await render(<AnimatedComponent animateFrom={animateFrom} animateTo={animateTo} config={config} />);
+  await wait(waitTime);
+
+  const updates = updatesContainer.getUpdates();
+  const nativeUpdates = await updatesContainer.getNativeSnapshots();
+
+  return [updates, nativeUpdates];
+}
+
+describe('WithSpring snapshots 📸, test various configs', () => {
+  describe('Empty configuration', () => {
+    test.each([
+      [20, 300],
+      [300, 50],
+      [0, 150],
+      [150, 0],
+    ] as Array<[number, number]>)(
+      'Empty config, from ${0} to ${1}',
+      async ([animateFrom, animateTo]: [number, number]) => {
+        const [updates, nativeUpdates] = await getSnapshotUpdates(animateFrom, animateTo, {});
+        const snapshotName = `empty_${animateFrom}_${animateTo}`;
+        expect(updates).toMatchSnapshots(Snapshots[snapshotName as keyof typeof Snapshots]);
+        expect(updates).toMatchNativeSnapshots(nativeUpdates, true);
+      },
+    );
+  });
+
+  describe('Invalid configuration, test warning', () => {
+    test.warn(
+      'Invalid mass and stiffness, config is { mass: -40, stiffness: -400 }',
+      '[Reanimated] Invalid spring config, stiffness must be grater than zero but got -400, mass must be grater than zero but got -40',
+      async () => {
+        await render(<AnimatedComponent animateFrom={30} animateTo={300} config={{ mass: -40, stiffness: -400 }} />);
+      },
+    );
```

Comment:
Why did you choose `test.warn` syntax instead of something similar to Jest syntax? I mean:
```js
test('Invalid mass and stiffness, config is { mass: -40, stiffness: -400 }', async () => {
  const result = await render(<AnimatedComponent animateFrom={30} animateTo={300} config={{ mass: -40, stiffness: -400 }} />);
  expect(result).toThrow('[Reanimated] Invalid spring config...', mode=WARNING);
});
```

Reply:
In jest there are two ways to test errors, 
* `expect(() => doSomething() ).toThrow( ...)` [LINK](https://jestjs.io/docs/expect#tothrowerror)
* `test.failing` [LINK](https://jestjs.io/docs/api#testfailingname-fn-timeout)
I've chosen the second one since it seemed simpler. Also there is no in-built way to check console.warn, you just have to use some function tracker, but I believe that having sth similar to error tests would be beneficial